src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -5094,8 +5059,47 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         )
         silentResult match {
           case SilentResultValue(fun1) =>
+            def transformRassoc(applied: Tree): Tree = {
+              val Apply(fn, arg :: Nil) = applied: @unchecked
+              val isByName = fn.symbol.paramLists.head.head.isByNameParam
+              if (isByName) applied
+              else {
+                def isRassocArg(t: Tree) = t.hasAttachment[RightAssociativeArg.type]
+                def checkRassocArg(t: Tree): Boolean = isRassocArg(t) && { t.removeAttachment[RightAssociativeArg.type]; true }
+                def rewriteRightAssoc(arg: Tree): Tree = {
+                  val vsym = context.owner.newValue(freshTermName(nme.RIGHT_ASSOC_OP_PREFIX), arg.pos.focus, FINAL | SYNTHETIC | ARTIFACT)
+                  vsym.setInfo(arg.tpe)
+                  val vdef = atPos(arg.pos) { ValDef(vsym, arg) setType NoType }
+                  context.pendingStabilizers ::= vdef
+                  arg.changeOwner(context.owner -> vsym)
+                  Ident(vsym).setType(singleType(NoPrefix, vsym)).setPos(arg.pos.focus)
+                }
+                def needsRewrite(t: Tree) = !treeInfo.isStableIdentifier(t, allowVolatile = false) && !treeInfo.isExprSafeToInline(t)
+                val rewriter = new Transformer {
```

Comment:
Do we need a recursive implementation? We know we have an Apply whose arg needs to go into a new local, seems that could be done by matching?